### DISCUSSION TITLE
位运算实验（上） | Super Blog

### DISCUSSION THREAD

**Discussion by giscus** [SuperPung/Blog#16] [General]
# 位运算实验（上） | Super Blog

Data Lab: Manipulating Bits #1 - 对于实验一（Data Lab）的一些想法（上）。

https://superpung.com/data-lab-1/

**Comment by sdythmh** [comment]
~((~x + (1 << n)) >> n) + 1 = ~(~x >> n + 1) + 1，这句话有点小错，由于+比>>的运算符优先级高，建议加上括号，改成~((~x + (1 << n)) >> n) + 1 = ~((~x >> n) + 1) + 1，后面亦如此
